src/consumer.cpp

```diff
@@ -254,19 +254,33 @@ MessageList Consumer::poll_batch(size_t max_batch_size, milliseconds timeout) {
 }
 
 Queue Consumer::get_main_queue() const {
+#if RD_KAFKA_VERSION <= 0x000b0500
     Queue queue(Queue::make_non_owning(rd_kafka_queue_get_main(get_handle())));
+#else
+    Queue queue(rd_kafka_queue_get_main(get_handle()));
+#endif
     queue.disable_queue_forwarding();
     return queue;
 }
 
 Queue Consumer::get_consumer_queue() const {
+#if RD_KAFKA_VERSION <= 0x000b0500
```

Comment:
I think this should do `if (rd_kafka_version() <= ...)`. There's an issue here if you build the library on a system that has rdkafka >= 0.11.5 but you deploy it on one that has rdkafka < 0.11.5. In that case, this won't really work.

Reply:
Also I'd probably create a function within this cpp file that gets an rdkafka queue pointer and returns a `Queue` so you don't have the exact same check deplicated N times.

Reply:
Ok, runtime check works too, you're right, it's safer. I'll wrap this up as per your comments.